tests/libvmi/vmi.go

```diff
@@ -135,3 +135,17 @@ func baseVmi(name string) *v1.VirtualMachineInstance {
 	}
 	return vmi
 }
+
+// WithMachineType specifies the vmi machine type.
+func WithMachineType(machineType string) Option {
```

Comment:
Defining good APIs is a difficult art because it is quite expensive to fix an API after it has become public. Hence it is better to keep functions (and everything else) local to a file, unless we are sure that it is needed in multiple places.

Since only one test wants to change `vmi.Spec.SchedulerName` it would be prudent to place `withScheduler` next to its only user.

P.S I see that k8s client-go calls a very similar function `WithSchedulerName`, so I would use `withSchedulerName` in KubeVirt.

Reply:
Hey @dankenigsberg!
I recommend you to look at the conversation on the same topic but on a different PR: https://github.com/kubevirt/kubevirt/pull/8075#discussion_r919994051.

Here's what I wrote there:
> I fundamentally disagree with this approach.
>
> I see `libvmi` as a package that we eventually aspire to be completely external to Kubevirt. IMHO, it's final goal is to serve users that want to easily build VMs, for either testing or production. Therefore, I don't think we should base our decisions of which options should be included on how Kubevirt is using it. We are just another user.
>
> The problem with utils is not that we have a "pile of stuff", is that it's unstructured, messy, ugly and unmaintainable. The purpose of libvmi to is to serve the same functionality but with patterns (e.g. builders) that would be clear, maintainable and easy to use.
>
> If `libvmi` is supposed to fit many use cases and many users I don't see why not have a "pile of stuff" as long as it's generic and not coupled to a specific use-case. If we implement such an option that probably would fit external users - why make them reinvent the same wheel?